test/historical-query.spec.ts

```diff
@@ -80,22 +116,9 @@ describe("rules from AJO", () => {
             consequences: [CONSEQUENCE],
           },
         ],
-      }).execute({
-        events: {
-          display: {
-            "6cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa":
-              {
-                event: {
-                  type: "display",
-                  id: "6cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa",
-                },
-                timestamp: 1681321319855,
-                count: 1,
-              },
-          },
-        },
-      })
+      }).execute(queryEventInIndexedDB(db, eventContext))
```

Comment:
It seems like you are trying to preserve the original functionality while adding IndexedDB too.  But that's not the requirement.  You need to forget about how events used to work and implement it to support events as an IndexedDB instance passed in on the context only.  We will no longer support accepting events as a JSON object.

```js
}).execute({events:db})
```

☝️  you will have to change the implementation in `historical.ts` to support this .